### PR TITLE
fix(flex): fix flex:1 failed when style minHeigh/minWidth

### PR DESCRIPTION
**Pull request by pba-cra**
fix:https://github.com/openwebf/webf/issues/27

### CODE REVIEW

File: webf/lib/src/rendering/flex.dart [hunk 1]
```diff
@@ -1164,7 +1164,13 @@ class RenderFlexLayout extends RenderLayoutBox {
           marginHorizontal = child.renderStyle.marginLeft.computedValue + child.renderStyle.marginRight.computedValue;
           marginVertical = child.renderStyle.marginTop.computedValue + child.renderStyle.marginBottom.computedValue;
           if (flexBasis != null) {
-            childSpace = flexBasis;
+            // Flexbox flexBasis == 0.0, need use minContentHeight/minContentWidth as childSpace
+            if(flexBasis == 0) {
```
Comment (andycall) [inline webf/lib/src/rendering/flex.dart:1168]: code style.

File: webf/lib/src/rendering/flex.dart [hunk 2]
```diff
@@ -1176,6 +1182,13 @@ class RenderFlexLayout extends RenderLayoutBox {
       // Flexbox with no size on main axis should adapt the main axis size with children.
       double initialFreeSpace = isMainSizeDefinite ? maxMainSize - totalSpace : 0;
 
+      double layoutContentMainSize = _isHorizontalFlexDirection ? boxSize!.width : boxSize!.height;
+
+      // Flexbox with minSize on main axis when layoutContentMainSize < minSize, adapt freeSpace
+      if(layoutContentMainSize <= _getMinMainAxisSize(this)  && initialFreeSpace == 0) {
```
Comment (andycall) [inline webf/lib/src/rendering/flex.dart:1188]: code style.

**Comment by andycall** [comment]
test spec missing?

**Comment by andycall** [comment]
Test spec required. @pba-cra

**Review by andycall — CHANGES_REQUESTED** [review]
Test spec required. @pba-cra

File: webf/lib/src/css/values/length.dart [hunk 3]
```diff
@@ -550,7 +550,7 @@ class CSSLength {
       }
     }
 
-    if (value == 0) {
+    if (value == 0 && propertyName != FLEX_BASIS) {
```
Comment (andycall) [inline webf/lib/src/css/values/length.dart:553]: CSSLengthValue should not depends on flex-basic property.

**Review by andycall — CHANGES_REQUESTED** [review]

**Review by devjiangzhou — APPROVED** [review]

**Review by andycall — APPROVED** [review]

**Pull request merged**